### DISCUSSION TITLE
PermissionError: [Errno 13] Permission denied: 'C:\\my\\path\\here'

### DISCUSSION THREAD

**Discussion by andreyrisukhin** [pyinstaller/pyinstaller#5468] [Help] [answered]
Hello! I'd like some help resolving an error I get when running pyinstaller to package my project.

I'm running pyinstaller in a miniconda3 prompt, and get quite a bit of output (up to 5359 INFO), after which there is a traceback and the error message `PermissionError: [Errno 13] Permission denied: 'C:\\my\\path\\here'`

Looking online, I need to run the prompt as an admin, but I am not sure how to do this with miniconda3. Is there a solution you can recommend?

Thank you!

**Comment by bwoodsend** [comment, marked as answer]
No, that's a really misleading Windows error. You should never need admin for PyInstaller. `PermissionError: [Errno 13] Permission denied:` actually means any of:
- You tried to read or write a file which already exists but is a folder (usually that just means delete the folder that's in the way).
- You tried to move or replace a file which is open in another application. This is very common in PyInstaller if you open a command prompt or file explorer inside the `dist` folder, then try to rebuild your program. PyInstaller wants to replace the contents of `dist` but it's already open in your prompt/explorer. In which case, close the offending programs.

If you give us the full error traceback and the command you ran to get it then I can tell you which case it is and how to solve.

**Reply by andreyrisukhin** [reply]
Thank you very much! Here is the full output of the command I ran (I am in a conda virtual environment in Python 3.7, and I pip installed pyinstaller in DAIS Repositories, while I am running the pyinstaller command in a subfolder):

`(vmDAIS37) C:\Users\Andre\Desktop\DAIS Repositories\Deep-Tracer-2.0>pyinstaller deeptracer2

109 INFO: PyInstaller: 4.1

109 INFO: Python: 3.7.9 (conda)

111 INFO: Platform: Windows-10-10.0.18362-SP0

111 INFO: wrote C:\Users\Andre\Desktop\DAIS Repositories\Deep-Tracer-2.0\deeptracer2.spec

111 INFO: UPX is not available.

125 INFO: Extending PYTHONPATH with paths
['C:\\Users\\Andre\\Desktop\\DAIS Repositories\\Deep-Tracer-2.0',
 'C:\\Users\\Andre\\Desktop\\DAIS Repositories\\Deep-Tracer-2.0']

140 INFO: checking Analysis

140 INFO: Building Analysis because Analysis-00.toc is non existent

140 INFO: Initializing module dependency graph...

156 INFO: Caching module graph hooks...

171 INFO: Analyzing base_library.zip ...

2796 INFO: Processing pre-find module path hook distutils from 'c:\\users\\andre\\miniconda3\\envs\\vmdais37\\lib\\site-packages\\PyInstaller\\hooks\\pre_find_module_path\\hook-distutils.py'.

2808 INFO: distutils: retargeting to non-venv dir 'c:\\users\\andre\\miniconda3\\envs\\vmdais37\\lib'

4640 INFO: Caching module dependency graph...

4843 INFO: running Analysis Analysis-00.toc

4843 INFO: Adding Microsoft.Windows.Common-Controls to dependent assemblies of final executable
  required by c:\users\andre\miniconda3\envs\vmdais37\python.exe

5359 INFO: Analyzing C:\Users\Andre\Desktop\DAIS Repositories\Deep-Tracer-2.0\deeptracer2
Traceback (most recent call last):
  File "c:\users\andre\miniconda3\envs\vmdais37\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\andre\miniconda3\envs\vmdais37\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\Andre\miniconda3\envs\vmDAIS37\Scripts\pyinstaller.exe\__main__.py", line 7, in <module>
    from deeptracer2.postprocessing.evaluation import evaluate_prediction
  File "c:\users\andre\miniconda3\envs\vmdais37\lib\site-packages\PyInstaller\__main__.py", line 114, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "c:\users\andre\miniconda3\envs\vmdais37\lib\site-packages\PyInstaller\__main__.py", line 65, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "c:\users\andre\miniconda3\envs\vmdais37\lib\site-packages\PyInstaller\building\build_main.py", line 720, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "c:\users\andre\miniconda3\envs\vmdais37\lib\site-packages\PyInstaller\building\build_main.py", line 667, in build
    exec(code, spec_namespace)
  File "C:\Users\Andre\Desktop\DAIS Repositories\Deep-Tracer-2.0\deeptracer2.spec", line 17, in <module>
    noarchive=False)
  File "c:\users\andre\miniconda3\envs\vmdais37\lib\site-packages\PyInstaller\building\build_main.py", line 242, in __init__
    self.__postinit__()
  File "c:\users\andre\miniconda3\envs\vmdais37\lib\site-packages\PyInstaller\building\datastruct.py", line 160, in __postinit__
    self.assemble()
  File "c:\users\andre\miniconda3\envs\vmdais37\lib\site-packages\PyInstaller\building\build_main.py", line 413, in assemble
    priority_scripts.append(self.graph.run_script(script))
  File "c:\users\andre\miniconda3\envs\vmdais37\lib\site-packages\PyInstaller\depend\analysis.py", line 304, in run_script
    pathname)
  File "c:\users\andre\miniconda3\envs\vmdais37\lib\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1407, in run_script
    with open(pathname, 'rb') as fp:
PermissionError: [Errno 13] Permission denied: 'C:\\Users\\Andre\\Desktop\\DAIS Repositories\\Deep-Tracer-2.0\\deeptracer2'`

Let me know if there is anything else I can do to help you help me :)

**Reply by bwoodsend** [reply]
The deeptracer2 in `pyinstaller deeptracer2` should be a Python file. It looks like it's a folder?

**Reply by andreyrisukhin** [reply]
Ah, I made that mistake. The file structure for this project is a folder Deep-Tracer-2.0 with the main.py and setup.py, and subfolders including deeptracer2 which contains algorithm specific information. Is there a better way for me to package a large project like this?

**Reply by bwoodsend** [reply]
Just `pyinstaller main.py` is probably what you want. It should pick up `deeptracer2` automatically the same as any library you use provided it's importable (test with `python -c import deeptracer2`) and `main.py` references it somewhere (contains `import deeptracer2` somewhere).

**Reply by andreyrisukhin** [reply]
Alright, I'll give that a go! Thank you so much for your time, I'm learning a lot from your explanation :)

**Reply by mrunalilipte4** [reply]
here, is code for image caption generator using Flickr dataset... Can you please help me out with this error???



import string
import numpy as np
from PIL import Image
import os
from pickle import dump, load
import numpy as np

from keras.applications.xception import Xception, preprocess_input
from keras.preprocessing.image import load_img, img_to_array
from keras.preprocessing.text import Tokenizer
from keras.preprocessing.sequence import pad_sequences
from keras.utils import to_categorical
from keras.layers.merge import add
from keras.models import Model, load_model
from keras.layers import Input, Dense, LSTM, Embedding, Dropout

# small library for seeing the progress of loops.
from tqdm.notebook import tqdm_notebook as tqdm
tqdm().pandas()


# Loading a text file into memory
def load_doc(filename):
    # Opening the file as read only
    file = open(filename, 'r')
    text = file.read()
    file.close()
    return text

# get all imgs with their captions
def all_img_captions(filename):
    file = load_doc(filename)
    captions = file.split('\n')
    descriptions ={}
    for caption in captions[:-1]:
        img, caption = caption.split('\t')
        if img[:-2] not in descriptions:
            descriptions[img[:-2]] = [caption]
        else:
            descriptions[img[:-2]].append(caption)
    return descriptions

##Data cleaning- lower casing, removing puntuations and words containing numbers
def cleaning_text(captions):
    table = str.maketrans('','',string.punctuation)
    for img,caps in captions.items():
        for i,img_caption in enumerate(caps):

            img_caption.replace("-"," ")
            desc = img_caption.split()

            #converts to lower case
            desc = [word.lower() for word in desc]
            #remove punctuation from each token
            desc = [word.translate(table) for word in desc]
            #remove hanging 's and a 
            desc = [word for word in desc if(len(word)>1)]
            #remove tokens with numbers in them
            desc = [word for word in desc if(word.isalpha())]
            #convert back to string

            img_caption = ' '.join(desc)
            captions[img][i]= img_caption
    return captions

def text_vocabulary(descriptions):
    # build vocabulary of all unique words
    vocab = set()
    
    for key in descriptions.keys():
        [vocab.update(d.split()) for d in descriptions[key]]
    
    return vocab

#All descriptions in one file 
def save_descriptions(descriptions, filename):
    lines = list()
    for key, desc_list in descriptions.items():
        for desc in desc_list:
            lines.append(key + '\t' + desc )
    data = "\n".join(lines)
    file = open(filename,"w")
    file.write(data)
    file.close()

dataset_text = "C:/Users/Dell/anaconda3/envs/testenv/Flickr8k_text"
dataset_images = "C:/Users/Dell/anaconda3/envs/testenv/Flickr8k_Dataset"
#dataset_feature="C:/Users/Dell/anaconda3/envs/testenv"

#we prepare our text data
filename = dataset_text + "/" + "Flickr8k.token.txt"
#loading the file that contains all data
#mapping them into descriptions dictionary img to 5 captions
descriptions = all_img_captions(filename)
print("Length of descriptions =" ,len(descriptions))

#cleaning the descriptions
clean_descriptions = cleaning_text(descriptions)

#building vocabulary 
vocabulary = text_vocabulary(clean_descriptions)
print("Length of vocabulary = ", len(vocabulary))

#saving each description to file 
save_descriptions(clean_descriptions, "descriptions.txt")


def extract_features(directory):
        model = Xception( include_top=False, pooling='avg' )
        features = {}
        for img in tqdm(os.listdir(directory)):
            filename = directory + "/" + img
            image = Image.open(filename)
            image = image.resize((299,299))
            image = np.expand_dims(image, axis=0)
            #image = preprocess_input(image)
            image = image/127.5
            image = image - 1.0
            
            feature = model.predict(image)
            features[img] = feature
        return features

#2048 feature vector
features = extract_features(dataset_images)
dump(features, open("features.p","wb"))



____________________________________________________________________________________________________________________________
PermissionError                           Traceback (most recent call last)
<ipython-input-46-c323c1b51dbd> in <module>
      1 #2048 feature vector
----> 2 features = extract_features(dataset_images)
      3 dump(features, open("features.p","wb"))

<ipython-input-45-8ced593b6a79> in extract_features(directory)
      4         for img in tqdm(os.listdir(directory)):
      5             filename = directory + "/" + img
----> 6             image = Image.open(filename)
      7             image = image.resize((299,299))
      8             image = np.expand_dims(image, axis=0)

~\anaconda3\lib\site-packages\PIL\Image.py in open(fp, mode, formats)
   2889 
   2890     if filename:
-> 2891         fp = builtins.open(filename, "rb")
   2892         exclusive_fp = True
   2893 

PermissionError: [Errno 13] Permission denied: 'C:/Users/Dell/anaconda3/envs/testenv/Flickr8k_Dataset/Flicker8k_Dataset'

**Reply by bwoodsend** [reply]
Looks like you're trying to read a folder as a file. `C:/Users/Dell/anaconda3/envs/testenv/Flickr8k_Dataset/Flicker8k_Dataset` certainly doesn't look like the name of an image.

**Reply by noahbroyles** [reply]
Thanks @bwoodsend! This was helpful.

**Reply by Likhithsai2580** [reply]
@bwoodsend I Have Error while using pyinstaller
Error Code
https://paste.pythondiscord.com/datavoreku.py

**Reply by bwoodsend** [reply]
That looks more likely to be antivirus related but I can't tell with just a stacktrace. Try disabling MacAfee and MS Defender and if that doesn't help then raise a new issue and provide a complete example code.

**Reply by Likhithsai2580** [reply]
Bro i disabled Avast And tried again this error came up

**Reply by bwoodsend** [reply]
Like I say, I can't tell with just a stacktrace. Raise a new issue and provide a complete example code.

**Reply by malvinjohnny** [reply]
```python
def uploadToBlobStorage(file_path,file_name):
   blob_service_client = BlobServiceClient.from_connection_string(connection_string)
   blob_client = blob_service_client.get_blob_client(container=container_name, blob=file_name)
   with open(file_path,"rb") as data:
      blob_client.upload_blob(data)
   print(f"Uploaded {file_name}.")
# calling a function to perform upload
uploadToBlobStorage('E:/azure','D0002.dcm')
```


I'm facing the same challenge and getting this:

```python
(env) PS J:\third_part> python run.py
Traceback (most recent call last):
  File "J:\third_part\run.py", line 1, in <module>
    from huk import  app
  File "J:\third_part\huk\__init__.py", line 31, in <module>
    from huk import  routes
  File "J:\third_part\huk\routes.py", line 258, in <module>
    uploadToBlobStorage('E:/azure','D0002.dcm')
  File "J:\third_part\huk\routes.py", line 254, in uploadToBlobStorage
    with open(file_path,"rb") as data:
PermissionError: [Errno 13] Permission denied: 'E:/azure'
```

**Reply by bwoodsend** [reply]
Presumably `E:/azure` is a directory - not a file.

**Reply by manjU-hash** [reply]
import logging
import os
import uuid
from datetime import datetime
import os
import pandas as pd
from datetime import datetime, timedelta

def process_files(hotel_name):
            # Convert the excel file to csv for ingestion
            ts = datetime.now()
            hotel_name = hotel_name.replace(' ', '_')
            filename = 'data_{0}_{1}.csv'.format(hotel_name, ts.strftime('%Y%m%d_%H%M%S'))

            process_file(final_filename=filename,download_path=r"C:/users/HP/OneDrive/Desktop",tz_obj=None)
            
def process_file(final_filename, download_path, tz_obj):
  
        # Read the files from the directory
        filenames = os.listdir(download_path)

        rename_dict = {
            'Occupied': 'Room On the books',
            'Available': 'Rooms Available',
            'Revenue (nights)': 'Room Revenue',
            'Occupancy': 'Occupancy',
            'Customers': 'Adults'
        }
PermissionError: [Errno 13] Permission denied: 'C:/users/HP/OneDrive/Desktop\\All'

Please guide how to resolve

**Reply by bwoodsend** [reply]
Can't tell without a proper traceback and I suspect that `process_file()` is incomplete given that it doesn't process any files but I'd be surprised if it's not the same issue - you're trying to read/write to a directory.

**Comment by malvinjohnny** [comment]
I look into that thank you

**Comment by therealbdawg** [comment]
hey guys... having an issue here... wondering if anyone has any ideas 
```
Running auto-py-to-exe v2.27.0
Building directory: C:\Users\BDAWG~1.LAP\AppData\Local\Temp\tmpyqt33663
Provided command: pyinstaller --noconfirm --onedir --console  "C:\Users\bdawg.LAPTOP-2CKTEBU7\Bot Business\Dentist Customer"
Recursion Limit is set to 5000
Executing: pyinstaller --noconfirm --onedir --console C:\Users\bdawg.LAPTOP-2CKTEBU7\Bot Business\Dentist Customer --distpath C:\Users\BDAWG~1.LAP\AppData\Local\Temp\tmpyqt33663\application --workpath C:\Users\BDAWG~1.LAP\AppData\Local\Temp\tmpyqt33663\build --specpath C:\Users\BDAWG~1.LAP\AppData\Local\Temp\tmpyqt33663

238375 INFO: PyInstaller: 5.4.1
238386 INFO: Python: 3.10.9
238401 INFO: Platform: Windows-10-10.0.22621-SP0
238418 INFO: wrote C:\Users\BDAWG~1.LAP\AppData\Local\Temp\tmpyqt33663\Dentist Customer.spec
238435 INFO: UPX is not available.
238450 INFO: Extending PYTHONPATH with paths
['C:\\Users\\bdawg.LAPTOP-2CKTEBU7\\Bot Business']
239589 INFO: checking Analysis
239593 INFO: Building Analysis because Analysis-04.toc is non existent
239609 INFO: Reusing cached module dependency graph...
239649 INFO: Caching module graph hooks...
239665 WARNING: Several hooks defined for module 'numpy'. Please take care they do not conflict.
239796 INFO: running Analysis Analysis-04.toc
239806 INFO: Adding Microsoft.Windows.Common-Controls to dependent assemblies of final executable
  required by C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2544.0_x64__qbz5n2kfra8p0\python.exe
239821 WARNING: lib not found: api-ms-win-appmodel-runtime-l1-1-0.dll dependency of C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.10_3.10.2544.0_x64__qbz5n2kfra8p0\python.exe
240051 INFO: Analyzing C:\Users\bdawg.LAPTOP-2CKTEBU7\Bot Business\Dentist Customer
An error occurred while packaging
Traceback (most recent call last):
  File "C:\Users\bdawg.LAPTOP-2CKTEBU7\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\auto_py_to_exe\packaging.py", line 131, in package
    run_pyinstaller()
  File "C:\Users\bdawg.LAPTOP-2CKTEBU7\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\PyInstaller\__main__.py", line 179, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "C:\Users\bdawg.LAPTOP-2CKTEBU7\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\PyInstaller\__main__.py", line 60, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "C:\Users\bdawg.LAPTOP-2CKTEBU7\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\PyInstaller\building\build_main.py", line 962, in main
    build(specfile, distpath, workpath, clean_build)
  File "C:\Users\bdawg.LAPTOP-2CKTEBU7\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\PyInstaller\building\build_main.py", line 884, in build
    exec(code, spec_namespace)
  File "C:\Users\BDAWG~1.LAP\AppData\Local\Temp\tmpyqt33663\Dentist Customer.spec", line 7, in <module>
    a = Analysis(
  File "C:\Users\bdawg.LAPTOP-2CKTEBU7\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\PyInstaller\building\build_main.py", line 409, in __init__
    self.__postinit__()
  File "C:\Users\bdawg.LAPTOP-2CKTEBU7\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\PyInstaller\building\datastruct.py", line 173, in __postinit__
    self.assemble()
  File "C:\Users\bdawg.LAPTOP-2CKTEBU7\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\PyInstaller\building\build_main.py", line 572, in assemble
    priority_scripts.append(self.graph.add_script(script))
  File "C:\Users\bdawg.LAPTOP-2CKTEBU7\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\PyInstaller\depend\analysis.py", line 268, in add_script
    self._top_script_node = super().add_script(pathname)
  File "C:\Users\bdawg.LAPTOP-2CKTEBU7\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\PyInstaller\lib\modulegraph\modulegraph.py", line 1415, in add_script
    with open(pathname, 'rb') as fp:
PermissionError: [Errno 13] Permission denied: 'C:\\Users\\bdawg.LAPTOP-2CKTEBU7\\Bot Business\\Dentist Customer'
```

**Reply by bwoodsend** [reply]
Looks like you gave autopytoexe `C:\Users\bdawg.LAPTOP-2CKTEBU7\Bot Business\Dentist Customer` as your script but it's a directory instead of a `.py` file.

**Reply by therealbdawg** [reply]
It was my first time making a python exe with multiple files i thought the onedir command was correct... figured it out haha. 
thanks

**Reply by bgmercury** [reply]
how did you figure it out

**Reply by therealbdawg** [reply]
Don't use onedir use the other command I forget what it is

**Comment by msitez** [comment]
hello guys... I am also having an same issue here... wondering if anyone has any ideas:

C:\windows\system32>pyinstaller -F [C:\Users\uif05360\PycharmProjects\pythonProject\appl curve.py]
1893 INFO: PyInstaller: 5.7.0
1893 INFO: Python: 3.9.13 (conda)
1978 INFO: Platform: Windows-10-10.0.19044-SP0
Traceback (most recent call last):
  File "C:\Users\uif05360\Anaconda3\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\uif05360\Anaconda3\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\uif05360\Anaconda3\Scripts\pyinstaller.exe\__main__.py", line 7, in <module>
  File "C:\Users\uif05360\Anaconda3\lib\site-packages\PyInstaller\__main__.py", line 194, in _console_script_run
    run()
  File "C:\Users\uif05360\Anaconda3\lib\site-packages\PyInstaller\__main__.py", line 178, in run
    spec_file = run_makespec(**vars(args))
  File "C:\Users\uif05360\Anaconda3\lib\site-packages\PyInstaller\__main__.py", line 54, in run_makespec
    spec_file = PyInstaller.building.makespec.main(filenames, **opts)
  File "C:\Users\uif05360\Anaconda3\lib\site-packages\PyInstaller\building\makespec.py", line 824, in main
    with open(specfnm, 'w', encoding='utf-8') as specfile:
PermissionError: [Errno 13] Permission denied: 'C:\\windows\\system32\\appl.spec'

**Reply by rokm** [reply]
Move out of the `c:\windows\system32` directory to user-writable working directory before you run PyInstaller.

**Reply by msitez** [reply]
how to move out of c:\windows\system32 directory, I tried to created a new folder in other disk, but the error was not eliminated:

```
C:\windows\system32>pyinstaller  --distpath D:\python_test\dist \python_test\appl curve.py
1894 INFO: PyInstaller: 5.7.0
1894 INFO: Python: 3.9.13 (conda)
1971 INFO: Platform: Windows-10-10.0.19044-SP0
Traceback (most recent call last):
  File "C:\Users\uif05360\Anaconda3\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\uif05360\Anaconda3\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\uif05360\Anaconda3\Scripts\pyinstaller.exe\__main__.py", line 7, in <module>
  File "C:\Users\uif05360\Anaconda3\lib\site-packages\PyInstaller\__main__.py", line 194, in _console_script_run
    run()
  File "C:\Users\uif05360\Anaconda3\lib\site-packages\PyInstaller\__main__.py", line 178, in run
    spec_file = run_makespec(**vars(args))
  File "C:\Users\uif05360\Anaconda3\lib\site-packages\PyInstaller\__main__.py", line 54, in run_makespec
    spec_file = PyInstaller.building.makespec.main(filenames, **opts)
  File "C:\Users\uif05360\Anaconda3\lib\site-packages\PyInstaller\building\makespec.py", line 824, in main
    with open(specfnm, 'w', encoding='utf-8') as specfile:
PermissionError: [Errno 13] Permission denied: 'C:\\windows\\system32\\appl.spec'
```

**Reply by bwoodsend** [reply]
`cd C:\Users\uif05360`

**Comment by PrinciplesKJ** [comment]
I faced the permission denied error and spent hours attempting various solutions. The issue wasn't related to folder permissions, root/admin access, or corrupted Python versions. Instead, the problem stemmed from the script's location (like some of the replies here suggested).

When using the "edit configuration" option (located at the top right of the main window), the interface prompts you for the script path in the first line. It's crucial NOT TO choose the folder itself but rather the FILE ITSELF in the tree view that appears. Otherwise, the system will attempt to access the entire folder instead of just the file. I hope this clarification helped.

**Comment by manikandanbalasubramani03** [comment]
Hi @bwoodsend 
I am getting below error in console could you help me out

PS C:\Users\manikandan\Desktop\mast-duplicate> python setup.py install
running install
error: can't create or remove files in install directory

The following error occurred while trying to add or remove files in the
installation directory:

    [Errno 13] Permission denied: 'C:\\Program Files\\Python310\\Lib\\site-packages\\test-easy-install-10172.write-test'

The installation directory you specified (via --install-dir, --prefix, or
the distutils default setting) was:

    C:\Program Files\Python310\Lib\site-packages\

Perhaps your account does not have write access to this directory?  If the
installation directory is a system-owned directory, you may need to sign in
as the administrator or "root" account.  If you do not have administrative
access to this machine, you may wish to choose a different installation
directory, preferably one that is listed in your PYTHONPATH environment
variable.

For information on other options, you may wish to consult the
documentation at:

  https://setuptools.readthedocs.io/en/latest/deprecated/easy_install.html

Please make the appropriate changes for your system and try again.

**Reply by bwoodsend** [reply]
That one looks like a genuine permission error caused by lack of permission. Anything in `C:\Program Files\` is typically owned by admin. Lots of ways of fixing it but I'd suggest you uninstall that Python environment, then reinstall it making sure that you **don't** install for all users.

**Reply by bwoodsend** [reply]
I'm not your personal free consultancy service. This has nothing to do with PyInstaller which is what this forum is for.

**Reply by manikandanbalasubramani03** [reply]
ok!

**Comment by hgeorgescu** [comment]
Run python as admin in your environment.

**Reply by bwoodsend** [reply]
No, please don't do that. Unless your Python code has a legitimate reason to be manipulating admin config or admin owned files, that's precisely what you avoid doing.

**Comment by hgeorgescu** [comment]
" don't do that" but I did it and it worked.
